### PR TITLE
MUMPFL-83 Search optionally by first name

### PR DESCRIPTION
**Pull request by vertein**
Adds in the ability to optionally search by first name.

Before UI:

image

After UI:
image

@jhanstra has graciously volunteered to work on MUMPFL-105 to update the UI when he is available.

### CODE REVIEW

File: my-profile-api/src/main/java/edu/wisc/my/profile/service/SearchUsersService.java [hunk 1]
```diff
@@ -2,15 +2,17 @@
...
+import edu.wisc.my.profile.model.SearchTerm;
 import edu.wisc.my.profile.model.User;
 
 public interface SearchUsersService {
     
     /**
      * Get a list of users based on search term
-     * @param searchTerm
-     * @return an alphabetized list based on last name, first name
+     * @param searchTerm -
```
Comment (apetro) [inline my-profile-api/src/main/java/edu/wisc/my/profile/service/SearchUsersService.java]: Looks like missing `@param` JavaDoc tags for `username` and `manifestGroups`.

File: my-profile-api/src/main/java/edu/wisc/my/profile/service/SearchUsersService.java [hunk 2]
```diff
@@ -2,15 +2,17 @@
...
+     * @return an alphabetized list based on last name, first name. Will return
+     * empty list if searchTerm is null or last name is empty
```
Comment (apetro) [inline my-profile-api/src/main/java/edu/wisc/my/profile/service/SearchUsersService.java]: :heart: the specificity of the return specification.

**Comment by apetro** [comment]
:+1:

File: my-profile-api/src/main/java/edu/wisc/my/profile/service/SearchUsersService.java (hunk 1 above)
Comment (vertein) [inline my-profile-api/src/main/java/edu/wisc/my/profile/service/SearchUsersService.java]: missing param indeed.  Cleaned it up

**Comment by timlevett** [comment]
:+1:

**Comment by jhanstra** [comment]
Added some labels and removed descriptive text: 

image

**Comment by apetro** [comment]
:+1:

**Comment by vertein** [comment]
Much simple so minimalist.  I'm glad you did this @jhanstra , I was caught in an endless loop with too much descriptive text and things were spiraling so much out of control.

:+1:

**Comment by jhanstra** [comment]
Haha, glad I could help @vertein

**Pull request merged**